spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -86,6 +90,7 @@ public class BigQueryTemplate implements BigQueryOperations {
   private final Logger logger = LoggerFactory.getLogger(BigQueryTemplate.class);
 
   private final int jsonWriterBatchSize;
+  private final ScheduledExecutorService executorService;
```

Comment:
Why not just `ExecutorService`? You're not using the scheduling features.
Also, the variable should probably be called `jsonWriterExecutorService` for clarity.